File: mods/mineos/utility.ts
namespace mineos {
  export function loadFiles(filesToLoad: string[]): void {
    const currentMod = minetest.get_current_modname()
    const currentDirectory = minetest.get_modpath(currentMod)
    for (const file of filesToLoad) {
      dofile(currentDirectory + "/" + file + ".lua")
    }
  }
  export function loadFile(file: string): any {
    const currentDirectory = minetest.get_modpath("mineos")
    return dofile(currentDirectory + "/" + file + ".lua")
  }

  vector.create = function(x?: number, y?: number, z?: number): Vec3 {
    let temp = vector.zero()
    temp.x = x || 0
    temp.y = y || 0
    temp.z = z || 0
    return temp
  };
  vector.create2d = function(x?: number, y?: number): Vec2 {
    let temp = {
      x: x || 0,
      y: y || 0
    }
    return temp
  }
  /**
   * Clamp a number between two number. (inclusive)
   * @param min Min value.
   * @param max Max value.
   * @param input Value to be clamped.
   * @returns Clamped value.
   */
  math.clamp = function(min: number, max: number, input: number): number {
    if (input < min) {
      return min
    } else if (input > max) {
      return max
    }
    return input
  }

  math.randomseed(os.time())
  const random = math.random;
  const gsub = string.gsub;
  const format = string.format
  // https://gist.github.com/jrus/3197011
  export function uuid(): string {
    const template ='xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'
    return gsub(template, "[xy]", (c: string) => {
      const v = (c == "x") && random(0, 0xf) || random(8, 0xb)
      return format("%x", v)
    })[0]
  }

  export const colorize = minetest.colorize;
}

File: mods/mineos/utility.lua
mineos = mineos or ({})
do
    function mineos.loadFiles(filesToLoad)
        local currentMod = minetest.get_current_modname()
        local currentDirectory = minetest.get_modpath(currentMod)
        for ____, file in ipairs(filesToLoad) do
            dofile(((currentDirectory .. "/") .. file) .. ".lua")
        end
    end
    function mineos.loadFile(file)
        local currentDirectory = minetest.get_modpath("mineos")
        return dofile(((currentDirectory .. "/") .. file) .. ".lua")
    end
    vector.create = function(x, y, z)
        local temp = vector.zero()
        temp.x = x or 0
        temp.y = y or 0
        temp.z = z or 0
        return temp
    end
    vector.create2d = function(x, y)
        local temp = {x = x or 0, y = y or 0}
        return temp
    end
    math.clamp = function(min, max, input)
        if input < min then
            return min
        elseif input > max then
            return max
        end
        return input
    end
    math.randomseed(os.time())
    local random = math.random
    local gsub = string.gsub
    local format = string.format
    function mineos.uuid()
        local template = "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx"
        return (gsub(
            template,
            "[xy]",
            function(c)
                local v = c == "x" and random(0, 15) or random(8, 11)
                return format("%x", v)
            end
        ))
    end
    mineos.colorize = minetest.colorize
end
